File: esports-backend/src/main/java/com/esports/dto/UserUpdateRequestDTO.java
package com.esports.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class UserUpdateRequestDTO {
    @Pattern(regexp = "^[a-zA-Z]+(\s[a-zA-Z]+)?$", message = "Name must contain only alphabets and at most one space")
    private String name;
    
    @Email(message = "Email must be valid")
    private String email;
    
    @Size(min = 6, message = "Password must be at least 6 characters")
    private String password;
}

File: esports-backend-dotnet/Upkart1/DTO/UserUpdateRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace Upkart1.DTO
{
    public class UserUpdateRequestDTO
    {
        [RegularExpression(@"^[a-zA-Z]+(\s[a-zA-Z]+)?$", ErrorMessage = "Name must contain only alphabets and at most one space")]
        public string? name { get; set; }

        [EmailAddress(ErrorMessage = "Email must be valid")]
        public string? email { get; set; }

        [StringLength(int.MaxValue, MinimumLength = 6, ErrorMessage = "Password must be at least 6 characters")]
        public string? password { get; set; }
    }
}